sympy/functions/elementary/exponential.py

```diff
@@ -195,7 +195,8 @@ def exp(arg, **kwargs):
 
     sympy.functions.elementary.exponential.log
     """
-    return Pow(S.Exp1, arg, **kwargs)
+    def __new__(cls, arg, **kwargs):
```

Comment:
If you would like to define ``exp`` as a class - probably you should use ``eval`` method, not this magic.  In that case ``exp`` will be also transformed to ``Pow`` automatically, but you can stop that with ``evaluate=False``.

Also, ``sqrt`` function must be adapted on the same way, maybe more.

Reply:
I wanted to preserve to ability to call something like `.subs(exp, sin)`. I've seen you don't have it in diofant.

Reply:
I'll try `eval`.

Reply:
> ability to call something like .subs(exp, sin)

This is irrelevant to my point.

Reply:
Even `.rewrite(exp)`? I think that's pretty useful.

Reply:
Doing a replace(exp, ...) requires `exp(x)` to be an instance of `exp`. I think to make that work, you'd need to do `class exp(Pow)`, which would be sort of a hybrid of what is there now and a pure "exp(x) is Pow(E, x)" approach.

Reply:
I mean, ``.subs(exp, sin)`` - is a different, unrelated issue.

> ``.rewrite(exp)?``

Yet another unrelated issue.

BTW, ``.rewrite`` works differently from ``.subs``.  For example, arguments doesn't ``sympify``ed, so you could keep this behaviour even if ``exp`` will be a Python function.  (Like it works for ``sqrt`` now.)

Reply:
> ``class exp(Pow)``

The tricky part here is that ``exp`` should be one-arg and ``Pow`` is two-arg.

That mean you can't use "subclassing-friendly" (c) way of rebuilding objects with ``self.func(*self.args)`` in ``Pow``'s methods.  And also you must use ``.base/exp`` properties here, not ``.args``.

I would suggest to test this approach on ``sqrt`` first.

Reply:
We already have to break `exp(x).args == (x,)`. `exp(x).args` would be `(E, x)`, and `exp(a, b)` would just return `Pow(a, b)` for compatibility. 

Alternately, we can do something like what's here, but define `__instancecheck__` on the `exp` class so that it works with `replace`. For `rewrite` it should work no matter what we do (even if `exp` is a function), because `rewrite` just does a lookup based on the name `arg._eval_rewrite_as_exp()`

I think I prefer the second idea (less hacky).

Reply:
``replace(exp, sin)`` will not work if ``len(exp(x).args) > 1``.